public/Instruction.js

```diff
@@ -0,0 +1,69 @@
+class Instruction {
+    mnemonic = undefined;
+    assembledLength = undefined;
+    assemble(program){
+        // Some stuff
+        return this.internalAssemble(program);
+    }
+    internalAssemble(program) {
+        return program;
+    }
+    disassemble(program){
+        // Some stuff
+        return this.internalDisassemble(program)
+    }
+    internalDisassemble(program) {
+        return program
+    }
+}
+
+class InstructionOperand {
+    constructor (type, value, startBit, endBit) {
+        this.type = type;
+        this.value = value;
+        this.startBit = startBit;
+        this.endBit = endBit;
+    }
+}
+
+class InstructionOpcode {
+    // flags can use the operand struct
+    constructor (mnemonic, flags, startBit, endBit) {
+        this.mnemonic = mnemonic;
+        this.flags = flags;
+        this.startBit = startBit;
+        this.endBit = endBit;
+    }
+}
+
+class Generic_Instruction extends Instruction {
+    mnemonic = undefined;
+    assembledLength = undefined;
+    machineFormat = [];
+    assemblyFormat = [];
+    constructor (mnemonic, assembledLength, mf, af) {
+        super ();
+        this.assembledLength = assembledLength;
+        this.mnemonic = mnemonic;
+        this.machineFormat = mf;
+        this.assemblyFormat = af;
+    }
+    internalAssemble(program) {
+        return [this.mnemonic, program.next_line()];
```

Comment:
Are you just doing this as a temp matter for now?

Reply:
Yes, the assemblyFormat has yet to be setup as a class... havent touched the actual assemble functions yet :)